fifteen_min/src/lib.rs

```diff
@@ -8,21 +10,29 @@ mod viewer;
 type App = map_gui::SimpleApp<()>;
 
 pub fn main() {
-    widgetry::run(
-        widgetry::Settings::new("15-minute neighborhoods").read_svg(Box::new(abstio::slurp_bytes)),
-        |ctx| {
-            map_gui::SimpleApp::new(ctx, map_gui::options::Options::default(), (), |ctx, app| {
-                vec![viewer::Viewer::random_start(ctx, app)]
-            })
-        },
-    );
+    let settings = Settings::new("15-minute neighborhoods");
+    run(settings);
+}
+
+fn run(mut settings: Settings) {
+    settings = settings.read_svg(Box::new(abstio::slurp_bytes));
+    widgetry::run(settings, |ctx| {
+        map_gui::SimpleApp::new(ctx, map_gui::options::Options::default(), (), |ctx, app| {
+            vec![viewer::Viewer::random_start(ctx, app)]
+        })
+    });
 }
 
 #[cfg(target_arch = "wasm32")]
 use wasm_bindgen::prelude::*;
 
 #[cfg(target_arch = "wasm32")]
-#[wasm_bindgen(start)]
-pub fn run() {
-    main();
+#[wasm_bindgen(js_name = "run")]
+pub fn run_wasm(root_dom_id: String, assets_base_url: String, assets_are_gzipped: bool) {
+    let settings = Settings::new("15-minute neighborhoods")
+        .root_dom_element_id(root_dom_id)
+        .assets_base_url(assets_base_url)
+        .assets_are_gzipped(assets_are_gzipped);
```

Comment:
I made these explicit and passed in via settings rather than trying to reverse engineer them from the window url. I think this is going to be easier to think about now and in the future as we hope to embed widgetry apps in less prescribed ways.

(e.g. maybe it won't be at the root of the domain, maybe the assets won't always be in the same place relative to the html file)